components/brave_sync/brave_profile_sync_service_impl.cc

```diff
@@ -812,6 +832,131 @@ void BraveProfileSyncServiceImpl::LoadSyncEntityInfo(
   }
 }
 
+bool BraveProfileSyncServiceImpl::IsOtherBookmarksFolder(
+    const jslib::SyncRecord* record) const {
+  auto bookmark = record->GetBookmark();
+  if (!bookmark.isFolder)
+    return false;
+
+  std::string other_node_object_id;
+  if (model_->other_node()->GetMetaInfo("object_id", &other_node_object_id) &&
+      record->objectId == other_node_object_id)
+    return true;
+
+  if (bookmark.order == tools::kOtherNodeOrder &&
+      bookmark.site.title == tools::kOtherNodeName &&
+      bookmark.site.customTitle == tools::kOtherNodeName) {
+    return true;
+  }
+
+  return false;
+}
+
+void BraveProfileSyncServiceImpl::ProcessOtherBookmarksFolder(
+    const jslib::SyncRecord* record,
+    bool* pass_to_syncer) {
+  std::string other_node_object_id;
+    // Save object_id for late joined desktop to catch up with current id
+    // iteration
+  if (!model_->other_node()->GetMetaInfo("object_id", &other_node_object_id) &&
+      record->action == jslib::SyncRecord::Action::A_CREATE) {
+    tools::AsMutable(model_->other_node())->SetMetaInfo("object_id",
+                                                        record->objectId);
+  } else {
+    // If late joined desktop has bookmarks in other_node before joining sync
+    // chain
+    if (other_node_object_id != record->objectId &&
+        other_node_object_id ==
+          tools::GenerateObjectIdForOtherNode(std::string())) {
+      tools::AsMutable(model_->other_node())->SetMetaInfo("object_id",
```

Comment:
What if  here local other_object_id will be of 2nd generation and remote other_object_id will be of 3rd generation?
We will not execute this statement. Is such case processed somewhere in the other place?

Reply:
Unfortunately, we can only handle first iteration vs future iteration because we have no obvious way to detect it unless we want to hash n time for each check or prepare a set of n iterations of id ahead.

Reply:
thanks, it makes sense

Reply:
after discussed with @bridiver. we should take remote id when it is different than current one. remote id is always the newer iteration, there won't be old iteration remote id because we always poll remote records before commit local changes, so out-of-date desktop will always catch up first

Reply:
fixed here https://github.com/brave/brave-core/pull/4404/commits/d4c494d0ae2d9c354a0be790403eda65c94c45bf